src/js/gallery.js

```diff
@@ -19,18 +36,19 @@
       replace:true,
       templateUrl:'gallery.html'
     };
-    
+
     function controller($scope){
       ionGalleryData.setGallery($scope.ionGalleryItems);
       ionGalleryData.setRowSize(parseInt($scope.ionGalleryRow));
-      
+
       var _drawGallery = function(){
         $scope.items = ionGalleryData.buildGallery();
         $scope.responsiveGrid = ionGalleryData.getGridSize();
+        $scope.labels = ionGalleryConfig.labels;
```

Comment:
Is there a reason for this tu be updated every time there's a change in the items array ? I prefer to move it outside that function, or is there a use case I'm not thinking of ?